From 861e10895bae6fdf80c72dd34badf9fc5495f92a Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 12 Aug 2025 20:52:56 +0100
Subject: [PATCH] modules: add slf4j-nop where it is missing

Default startup of elasticsearch results in warnings:

> [WARN ][stderr ] [es] SLF4J: No SLF4J providers were found.
> [WARN ][stderr ] [es] SLF4J: Defaulting to no-operation (NOP) logger implementation
> [WARN ][stderr ] [es] SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.

It seems like most modules and plugins install nop logger, but some
don't. If one installs them everywhere, these warnings seem to go
away.
---
 modules/repository-gcs/build.gradle   | 1 +
 modules/repository-s3/build.gradle    | 1 +
 x-pack/plugin/ent-search/build.gradle | 1 +
 3 files changed, 3 insertions(+)

diff --git a/modules/repository-gcs/build.gradle b/modules/repository-gcs/build.gradle
index 2a2efb9c4fb..21724d36c11 100644
--- a/modules/repository-gcs/build.gradle
+++ b/modules/repository-gcs/build.gradle
@@ -30,6 +30,7 @@ dependencies {
   runtimeOnly 'com.google.guava:failureaccess:1.0.1'
   api "commons-logging:commons-logging:${versions.commonslogging}"
   api "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
+  runtimeOnly "org.slf4j:slf4j-nop:${versions.slf4j}"
   api "commons-codec:commons-codec:${versions.commonscodec}"
   api 'com.google.api:api-common:2.3.1'
   api 'com.google.api:gax:2.20.1'
diff --git a/modules/repository-s3/build.gradle b/modules/repository-s3/build.gradle
index a8b80855009..d29231c4fea 100644
--- a/modules/repository-s3/build.gradle
+++ b/modules/repository-s3/build.gradle
@@ -46,6 +46,7 @@ dependencies {
   runtimeOnly "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
   runtimeOnly "org.reactivestreams:reactive-streams:${versions.reactive_streams}"
   runtimeOnly "org.slf4j:slf4j-api:${versions.slf4j}"
+  runtimeOnly "org.slf4j:slf4j-nop:${versions.slf4j}"
   runtimeOnly "software.amazon.awssdk:arns:${versions.awsv2sdk}"
   runtimeOnly "software.amazon.awssdk:aws-query-protocol:${versions.awsv2sdk}"
   runtimeOnly "software.amazon.awssdk:checksums-spi:${versions.awsv2sdk}"
diff --git a/x-pack/plugin/ent-search/build.gradle b/x-pack/plugin/ent-search/build.gradle
index 5955ba05f0e..9d2c48dc7c9 100644
--- a/x-pack/plugin/ent-search/build.gradle
+++ b/x-pack/plugin/ent-search/build.gradle
@@ -26,6 +26,7 @@ dependencies {
   implementation "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
   implementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
   implementation "org.slf4j:slf4j-api:${versions.slf4j}"
+  runtimeOnly "org.slf4j:slf4j-nop:${versions.slf4j}"
   implementation "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}"
   implementation "com.networknt:json-schema-validator:${versions.networknt_json_schema_validator}"
 
-- 
2.48.1

